controller/konnect/watch_kongkey.go

```diff
@@ -176,3 +184,43 @@ func enqueueKongKeyForKonnectControlPlane(
 		return ret
 	}
 }
+
+func enqueueKongKeyForKongKeySet(cl client.Client) handler.MapFunc {
+	return func(ctx context.Context, obj client.Object) []reconcile.Request {
+		keySet, ok := obj.(*configurationv1alpha1.KongKeySet)
+		if !ok {
+			return nil
+		}
+		var l configurationv1alpha1.KongKeyList
+		if err := cl.List(ctx, &l, &client.ListOptions{}); err != nil {
+			return nil
+		}
+
+		var ret []reconcile.Request
+		for _, key := range l.Items {
+			keySetRef := getKeySetRef(&key)
+			ref, ok := keySetRef.Get()
+			if !ok {
+				continue
+			}
+			if ref.Type != configurationv1alpha1.KeySetRefNamespacedRef {
+				ctrllog.FromContext(ctx).V(logging.DebugLevel.Value()).Info(
+					"unsupported KongKeySetRef for KongKey",
+					"KongKey", key, "refType", ref.Type,
+				)
+				continue
+			}
+			if ref.NamespacedRef == nil || ref.NamespacedRef.Name != keySet.Name {
+				continue
+			}
+			ret = append(ret, reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: key.Namespace,
+					Name:      key.Name,
+				},
+			})
+		}
+
+		return ret
```

Comment:
I believe all of this can be changed into a list using an index (which would have to be added) and a  [`objectListToReconcileRequests`](https://github.com/Kong/gateway-operator/blob/820f2c87a21189a8aa3157e573d8b0819de50a53/controller/konnect/watch.go#L112) call. WDYT?

Reply:
11c59a4

Reply:
👍